GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -58,8 +57,17 @@ public partial class GitIndexProjection : IDisposable, IProfilerOnlyIndexProject
 
         private BlobSizes blobSizes;
         private IPlaceholderCollection placeholderDatabase;
+
         private ISparseCollection sparseCollection;
         private SparseFolderData rootSparseFolder;
+        private HashSet<string> sparsePaths;
+
+        // The set of sparse folders that had child folders added or removed
+        // during the last projection refresh.  If there are placeholders
+        // on disk for these folders we will update their last write time to
+        // force a refresh of the folders in File Explorer.
+        private HashSet<string> sparseFoldersNeedingRefresh;

```

Comment:
What are your performance and memory numbers with this change?  Especially when there are a lot of folders?

Reply:
I will run some tests and post back with results, thanks!